### PR TITLE
Handling names for bundles: register (new command) and list (generalized previous one).

### PR DESCRIPTION
**Pull request by facundobatista**
Also took the opportunity for a couple of small improvements:

- empty line after commands help text is superfluous (the formatter takes care of that, so I changed those that were inconsistent)

- a `meta` was redundantly used in a argparse option

### CODE REVIEW

File: charmcraft/commands/store/__init__.py [hunk 1]
```diff
@@ -36,6 +36,10 @@
 
 logger = logging.getLogger('charmcraft.commands.store')
 
+# entity types
+CHARM = 'charm'
```
Comment (sergiusens) [inline charmcraft/commands/store/__init__.py:40]: Random suggestion (might need a better name that TYPES)

```python
TYPES = namedtuple("Types", "CHARM BUNDLE")(CHARM="charm", BUNDLE="bundle")
```

**Comment by sergiusens** [comment]
There are also conflicts

File: charmcraft/commands/store/__init__.py [hunk 2]
```diff
@@ -162,28 +163,64 @@ def fill_parser(self, parser):
...
+        release them for wider consumption.
+
+        Charmhub operates on the 'principle of least surprise' with regard
+        to naming. A bundle with a well-known name should provide the best
+        system for the service most people associate with that name.  Bundles
+        can be renamed in the Charmhub, but we would nonetheless ask
+        you to use a qualified name, such as `yourname-bundlename` if you are
+        in any doubt about your ability to meet that standard.
+
+        We discuss registrations in Charmhub Discourse:
```
[the remark this answers is not in the document]
Comment (facundobatista) [reply charmcraft/commands/store/__init__.py:189]: Fixed here and in the `register`'s help text too.

File: charmcraft/commands/store/__init__.py (hunk 1 above)
Comment (facundobatista) [reply charmcraft/commands/store/__init__.py:40]: I considered using an Enum, but it was overkilling (and didn't fit ok). Using a namedtuple here looks like a hack, but fits perfectly with the usage.

I'd drop the uppercases as this is not longer a "const", but more a class, same with attributes.

Improved! Thanks.

**Review by sergiusens — APPROVED** [review]

**Review by sergiusens — APPROVED** [review]

**Pull request merged**